yarl/_url.py

```diff
@@ -209,38 +210,50 @@ def __new__(
         else:
             raise TypeError("Constructor parameter should be str")
 
+        cache: Dict[str, Union[str, int, None]] = {}
```

Comment:
It would be better to make this a typed dict so any mistakes get caught early

Reply:
Probably something for a future PR though as I don't want to make this one larger